server/protocols/webadmin-integration-test/webadmin-integration-test-common/src/main/java/org/apache/james/webadmin/integration/WebAdminServerIntegrationContract.java

```diff
@@ -54,24 +54,22 @@
 
 import io.restassured.RestAssured;
 
-public abstract class WebAdminServerIntegrationTest {
+public interface WebAdminServerIntegrationContract {
 
-    private static final String DOMAIN = "domain";
-    protected static final String USERNAME = "username@" + DOMAIN;
-    private static final String USERNAME_2 = "username2@" + DOMAIN;
-    private static final String GROUP = "group@" + DOMAIN;
-    private static final String SPECIFIC_DOMAIN = DomainsRoutes.DOMAINS + SEPARATOR + DOMAIN;
-    private static final String SPECIFIC_USER = UserRoutes.USERS + SEPARATOR + USERNAME;
-    private static final String MAILBOX = "mailbox";
-    private static final String SPECIFIC_MAILBOX = SPECIFIC_USER + SEPARATOR + UserMailboxesRoutes.MAILBOXES + SEPARATOR + MAILBOX;
+    String DOMAIN = "domain";
+    String USERNAME = "username@" + DOMAIN;
+    String USERNAME_2 = "username2@" + DOMAIN;
+    String GROUP = "group@" + DOMAIN;
+    String SPECIFIC_DOMAIN = DomainsRoutes.DOMAINS + SEPARATOR + DOMAIN;
+    String SPECIFIC_USER = UserRoutes.USERS + SEPARATOR + USERNAME;
+    String MAILBOX = "mailbox";
+    String SPECIFIC_MAILBOX = SPECIFIC_USER + SEPARATOR + UserMailboxesRoutes.MAILBOXES + SEPARATOR + MAILBOX;
 
-    protected static final String ALIAS_1 = "alias1@" + DOMAIN;
-    protected static final String ALIAS_2 = "alias2@" + DOMAIN;
-
-    private DataProbe dataProbe;
+    String ALIAS_1 = "alias1@" + DOMAIN;
+    String ALIAS_2 = "alias2@" + DOMAIN;
 
     @BeforeEach
-    void setUp(GuiceJamesServer guiceJamesServer) throws Exception {
+    default void setUp(GuiceJamesServer guiceJamesServer, DataProbe dataProbe) throws Exception {
         dataProbe = guiceJamesServer.getProbe(DataProbeImpl.class);
```

Comment:
Don't you do that now in the `resolveParam` method of the builder? Is it possible to remove this line then?